### PR TITLE
docs - add module landing page for gp_sparse_vector

### PR DESCRIPTION
**Pull request by lisakowen**
we are including the greenplum contrib module gp_sparse_vector in the gpdb 6.0 distribution - in this PR:
- add doc content for the gp_sparse_vector module.  most of this info was taken from the README in gpcontrib/gp_sparse_vector, with some reorg and updates to commands so they would work in gpdb 6.

notes:
- there appears to be a similar, fuller-featured module in MADlib.  some datatypes and functions are the same, but MADlib looks to schema-qualifies its versions.
- MADlib docs also are using an extended version of the dictionary/word example.
- the gp_sparse_vector docs in this PR link to MADlib sparse vector module docs.

questions:
- when might someone choose to use gp_sparse_vector over MADlib?
- why include gp_sparse_vector when this module appears to be in MADlib in some form?

doc review site link:  http://docs-lisa-gpdb6.cfapps.io/6-0/ref_guide/modules/gp_sparse_vector.html

### CODE REVIEW

**Comment by Eulerizeit** [comment]
Sparse vectors was included in Greenplum to originally support MADlib. I don't know if they have their own implementation independent of this. 

It can be useful separate from MADlib as a means of storing sparse data in a single table such as storing a customer ID and all the items they purchased during one trip.

File: gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+        <p>This type of array arises often in text processing, where a dictionary may
+          have 40-100K terms and the number of words in a particular document is stored
+          in a vector. This array would occupy slightly more than 320KB of memory/disk,
+          most of it zeros. Any operation that you perform on this data will be doing
+          work on 40,001 fields that are not important.</p>
+        <p>The Greenplum Database built-in <codeph>array</codeph> datatype utilizes a
+          bitmap for null values, but it is a poor choice for this use case because
+          it is not optimized for <codeph>float8[]</codeph> or for long runs of
+          zeros instead of nulls, and the bitmap is not run-length-encoding- (RLE)
+          compressed. Even if we store each zero as <codeph>NULL</codeph> in the
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml]: we store each zero as > each zero were stored as a

File: gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+          array, the bitmap for nulls would have 5KB of bits in it to mark the
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml]: "5KB of bits in it" seems really confusing to me.  Should "of bits in it" just be removed?

File: gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+          module is named <codeph>svec</codeph>. <codeph>svec</codeph> supports only
+          <codeph>float8</codeph> vector values.</note>
+        <p>You can construct an <codeph>svec</codeph> directly from a float array as 
+          follows:</p>
+        <codeblock>SELECT ('{0, 13, 37, 53, 0, 71 }'::float8[])::svec;</codeblock>
+        <p>The <codeph>gp_sparse_vector</codeph> module supports the vector operators
+          <codeph>&lt;</codeph>, <codeph>&gt;</codeph>, <codeph>*</codeph>,
+          <codeph>**</codeph>, <codeph>/</codeph>, <codeph>=</codeph>,
+          <codeph>+</codeph>, <codeph>sum()</codeph>, <codeph>vec_count_nonzero()</codeph>,
+          etc. These operators take advantage of the efficient sparse storage format,
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml]: etc -> and so on

File: gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml [hunk 4]
```diff
@@ -0,0 +1,246 @@
...
+        <codeblock>#_times_term_appears_in_this_doc * log( #_docs / #_docs_the_term_appears_in )</codeblock>
+        <p><codeph>#_docs</codeph> is the total number of
+          documents (4 in this case). Note that there is one divisor for each
+          dictionary word and its value is the number of times that word appears in the
+          document.</p>
+        <p>For example, the term "document" in document 1 would have a weight of
+          <codeph>1 * log( 4/3 )</codeph>. In document 4, the term would have a weight
+          of <codeph>2 * log( 4/3 )</codeph>. Terms that appear in every document 
+          would have weight <codeph>0</codeph>.</p>
+        <p>This single vector for the whole corpus is then scalar product
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml]: then scalar -> then a scalar

File: gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+    AS foo, corpus ORDER BY docnum;
+ docnum |                                                                          tf_idf                                                                          
+--------+----------------------------------------------------------------------------------------------------------------------------------------------------------
+      1 | {0,0,0,0,0.693147180559945,0.287682072451781,0,0.693147180559945,0.693147180559945,0,0,0,1.38629436111989,0,0,0.287682072451781,0,1.38629436111989,0}
+      2 | {1.38629436111989,0,0,0,0.693147180559945,0.287682072451781,1.38629436111989,0.693147180559945,0,0,0,0,0,0,1.38629436111989,0.575364144903562,0,0,0}
+      3 | {0,0,1.38629436111989,1.38629436111989,0,0,0,0,0,0.693147180559945,1.38629436111989,1.38629436111989,0,1.38629436111989,0,0,0.693147180559945,0,1.38629436111989
+}
+      4 | {0,1.38629436111989,0,0,0,0.575364144903562,0,0,0.693147180559945,0.693147180559945,0,0,0,0,0,0.575364144903562,0.693147180559945,0,0}
+</codeblock>
+        <p>We can determine the <i>angular distance</i> between one document and the
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml]: We -> You

File: gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+     (SELECT tf_idf testdoc FROM weights WHERE docnum = 1 LIMIT 1) foo
+ORDER BY 1;
+
+ docnum | angular_distance 
+--------+------------------
+      1 |             0.00
+      2 |            78.82
+      3 |            90.00
+      4 |            80.02</codeblock>
+        <p>We can see that the angular distance between document 1 and itself is 0
```
Comment (dyozie) [inline gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml]: We -> You

**Comment by lisakowen** [comment]
thanks for the context, @Eulerizeit

File: gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml (hunk 4 above)
Comment (lisakowen) [reply gpdb-doc/dita/ref_guide/modules/gp_sparse_vector.xml]: i think "scalar product multiplied" is correct, doesn't need an **a**?  maybe some dashes?

**Pull request merged**